modules/sources/contracts/fetch-tokenpair-data.ts

```diff
@@ -255,6 +356,25 @@ function addBToAPriceCallsToMulticaller(
     });
 }
 
+function addBptBToAPriceCallsToMulticaller(
+    tokenPair: TokenPair,
+    balancerRouterAddress: string,
+    multicaller: ViemMulticallCall[],
+) {
+    if (tokenPair.tokenBIndex === undefined || tokenPair.poolTokensLength === undefined) {
```

Comment:
I would add vaultVersion to the tokenPair type instead of checking for undefined here

Reply:
This check is just to ensure to TS that tokenBIndex and poolTokensLength will not be undefined, all the tokenPairs will arrive with tokenBIndex and poolsTokensLength defined at this point, but for non-BPT token pairs these parameters can be undefined, so I made this verification instead of forcing "tokenBIndex as number"